design/ceph/ceph-jaeger-tracing.md

```diff
@@ -0,0 +1,71 @@
+---
+title: deploy jaeger tracing components to ceph cluster
+target-version: release-1.8
+---
+
+# Tracing - Jaeger
+
+## Summary
+
+In order to make it possible to trace in Ceph, we would need the components of jaeger tracing to be deployed in the ceph cluster.
+
+the jaeger components consists of:
+1. elasticsearch - needs to be manually deployed.
+2. jaeger collector, agent and UI - are all managed by the jaeger Operator.
+
+[Jaeger operator documentation](https://www.jaegertracing.io/docs/1.28/operator/)
+
+### Goals
+
+to be able to see traces from the ceph cluster (rgw and osd) in the jaeger UI.
+
+## Proposal details
+
+When Ceph is deployed via Rook, we would like to add a parameter to the rook CR that will indicate to enable tracing and deploy the jaeger resources.
+
+<b> Ceph cluster yaml file:
+```yaml
+  apiVersion: ceph.rook.io/v1
+  kind: CephCluster
+  metadata:
+    name: rook-ceph
+    namespace: rook-ceph
+    [...]
+  spec:
+    [...]
+    tracing:
+      enabled: true
+    [...]
+
+```
+<b> default jaeger CR:
+```yaml
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: jaeger-production
+spec:
```

Comment:
Is there something dynamic about this CR that would change depending on the cephcluster configuration? At a glance it looks very static and could just be created separately from Rook at the same time the jaeger operator is created.

Reply:
> 

not at the scope here. the main motivation here would be usability and user experience.

in the future, we may add multisite specific support, and use jaeger with the "streaming" deployment strategy.
meaning that the traces are sent from collectors located in the different sites to a centralized kafka message broker and taken from there to the db backend.
this would mean the deployment strategy may depend on whether we have multiple sites in rook.